PubSub/src/Subscription.php

```diff
@@ -845,6 +891,24 @@ private function formatSubscriptionDurations(array $options)
         return $options;
     }
 
+    /**
+     * Format dead letter topic subscription data for API.
+     *
+     * @param array $subscription
+     * @return array
+     */
+    private function formatDeadLetterPolicyForApi(array $subscription)
+    {
+        if (isset($subscription['deadLetterPolicy'])) {
+            if ($subscription['deadLetterPolicy']['deadLetterTopic'] instanceof Topic) {
+                $topic = $subscription['deadLetterPolicy']['deadLetterTopic'];
+                $subscription['deadLetterPolicy']['deadLetterTopic'] = $topic->name();
```

Comment:
What about maximum delivery attempts field?

Reply:
No transformation is required to make the max delivery attempts field compatible with the API.